src/SourceNodes/sourcenode.ts

```diff
@@ -79,6 +80,7 @@ abstract class SourceNode extends GraphNode {
         this._callbacks = [];
         this._renderPaused = false;
         this._displayName = TYPE;
+        this._id = src;
```

Comment:
this `_id` will not be unique.. we can have the same source in several nodes right?

Reply:
Good point! I'll re-use the unique id approach here instead.